src/main/java/uk/gov/companieshouse/officerfiling/api/service/FilingDataServiceImpl.java

```diff
@@ -66,34 +62,29 @@ public FilingDataServiceImpl(OfficerFilingService officerFilingService,
      */
     @Override
     public FilingApi generateOfficerFiling(String transactionId, String filingId, String ericPassThroughHeader) {
-        var filing = new FilingApi();
-        Optional<OfficerFiling> officerFiling = officerFilingService.get(filingId, transactionId);
-        if(officerFiling.isPresent()) {
-            OfficerFilingData presentOfficerFilingData = officerFiling.get().getData();
-            if (presentOfficerFilingData.getResignedOn() != null) {
-                //has a removal date so must be a TM01
-                filing.setKind("officer-filing#termination");
-                setTerminationFilingApiData(filing, transactionId, filingId, ericPassThroughHeader);
-            } else if (presentOfficerFilingData.getReferenceEtag() == null) {
-                //has no Etag (and has no removal date) so it must be an AP01
-                filing.setKind("officer-filing#appointment");
-                setAppointmentFilingApiData(filing, transactionId, filingId, ericPassThroughHeader);
-            } else {
-                throw new NotImplementedException("Kind cannot be calculated using given data for transaction " + transactionId );
-            }
-        } else {
+        final var filing = new FilingApi();
+        final var officerFiling = officerFilingService.get(filingId, transactionId)
+                .orElseThrow(() -> new IllegalStateException(String.format("Officer not found when generating filing for %s", filingId)));
+        final var presentOfficerFilingData = officerFiling.getData();
+
+        if (presentOfficerFilingData.getResignedOn() != null) {
+            // Has a removal date so must be a TM01
             filing.setKind("officer-filing#termination");
-            setTerminationFilingApiData(filing, transactionId, filingId, ericPassThroughHeader);
+            setTerminationFilingApiData(filing, transactionId, filingId, ericPassThroughHeader, officerFiling);
+        } else if (presentOfficerFilingData.getReferenceEtag() == null) {
+            // Has no Etag (and has no removal date) so it must be an AP01
+            filing.setKind("officer-filing#appointment");
+            setAppointmentFilingApiData(filing, transactionId, filingId, ericPassThroughHeader, officerFiling);
+        } else {
+            // Has an Etag (and has no removal date) so it must be an CH01
+            filing.setKind("officer-filing#update");
+            setUpdateFilingApiData(filing, transactionId, filingId, ericPassThroughHeader, officerFiling);
         }
```

Comment:
should we check `filing.getdirectorsDetailsChangedDate` and keep the last else block to throw NotImplementedException?